transmart-core-db/grails-app/services/org/transmartproject/db/clinical/MultidimensionalDataResourceService.groovy

```diff
@@ -116,6 +116,13 @@ class MultidimensionalDataResourceService implements MultiDimensionalDataResourc
     Dimension getDimension(String name) {
         DimensionDescription.findByName(name).dimension
     }
+    
+    DimensionImpl getDimensionForStudies(String dimensionName, Collection<MDStudy> studies) {
```

Comment:
Looking at this again, there is still a need for this call. But currently this call does two things: convert a string to a dimension object, and check that the the dimension is part of the accessible studies. It would be more clear to split those operations. Convert the string to dimension separately and then verify that the user can access them. So maybe you can change this to something like `Collection<Dimension> inaccessibleDimensions(Collection<Dimension> dimensions, User user)` that returns the dimensions to which the user does not have access and a wrapper `void verifyDimensionsAccessible(Collection<Dimension>, User user)` and move it them `AccessControlChecks`.